### PR TITLE
Refactored ModelNet dataset to use KaolinDataset

### PR DESCRIPTION
**Pull request by TommyX12**
- Fixed bugs in `KaolinDataset` base class
- Refactored `ModelNet` dataset to use `KaolinDataset`

### CODE REVIEW

File: kaolin/datasets/modelnet.py [hunk 1]
```diff
@@ -12,63 +12,62 @@
...
 import torch
 import os
 from glob import glob
-from tqdm import tqdm
 
 from kaolin.rep.TriangleMesh import TriangleMesh
-from kaolin.transforms import transforms as tfs
 
 
 class ModelNet(object):
```
Comment (Caenorst) [inline kaolin/datasets/modelnet.py]: It's not using KaolinDataset as a base class

File: kaolin/datasets/modelnet.py [hunk 2]
```diff
@@ -12,63 +12,62 @@
...
     """
 
-    def __init__(self, basedir: str,
-                 split: Optional[str] = 'train',
-                 categories: Optional[Iterable] = ['bed'],
-                 transform: Optional[Callable] = None,
-                 device: Optional[Union[torch.device, str]] = 'cpu'):
+    def initialize(self, root: str,
+                   split: Optional[str] = 'train',
+                   categories: Optional[Iterable] = ['bed']):
```
Comment (Caenorst) [inline kaolin/datasets/modelnet.py]: There shouldn't be any default value for categories.

On a side note, if you want to set a default argument value it shouldn't be a mutable value, it should be `None` and then the default value is set inside the function.

**Review by Caenorst — COMMENTED** [review]
Can you remove the files that you modified from .flake8 `exclude` list ?

File: kaolin/datasets/modelnet.py (hunk 1 above)
Comment (TommyX12) [reply kaolin/datasets/modelnet.py]: Done.

File: kaolin/datasets/modelnet.py (hunk 2 above)
Comment (TommyX12) [reply kaolin/datasets/modelnet.py]: Done.

Comment (Caenorst) [reply kaolin/datasets/modelnet.py]: An update, can you set all categories by default ?

**Comment by Caenorst** [comment]
Also just need to sign your commits, and you're good to go

**Review by Caenorst — APPROVED** [review]
LGTM

**Pull request merged**